mint-auditor/src/http_api/service.rs

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Mint auditor service for handling http requests
+
+use crate::{
+    db::{BlockAuditData, BlockBalance, Counters, MintAuditorDb},
+    http_api::api_types::{BlockAuditDataResponse, CountersResponse},
+    Error,
+};
+
+/// Service for handling auditor requests
+pub struct MintAuditorHttpService {}
+
+/// Service for handling auditor requests
+impl MintAuditorHttpService {
+    /// get counters
+    pub fn get_counters(mint_auditor_db: &MintAuditorDb) -> Result<CountersResponse, Error> {
+        let conn = mint_auditor_db.get_conn()?;
+        Ok(CountersResponse::from(&Counters::get(&conn)?))
+    }
+
+    /// Get the audit data for a target block
+    pub fn get_block_audit_data(
+        block_index: u64,
+        mint_auditor_db: &MintAuditorDb,
+    ) -> Result<BlockAuditDataResponse, Error> {
+        let conn = mint_auditor_db.get_conn()?;
+
+        let block_audit_data = BlockAuditData::get(&conn, block_index)?;
+
+        let balances = BlockBalance::get_balances_for_block(&conn, block_audit_data.block_index())?;
+
+        Ok(BlockAuditDataResponse::new(block_audit_data, balances))
+    }
+
+    /// Get the audit data for the last synced block.
+    pub fn get_last_block_audit_data(
+        mint_auditor_db: &MintAuditorDb,
+    ) -> Result<BlockAuditDataResponse, Error> {
+        let conn = mint_auditor_db.get_conn()?;
+
+        let block_audit_data =
+            BlockAuditData::last_block_audit_data(&conn)?.ok_or(Error::NotFound)?;
+
+        let balances = BlockBalance::get_balances_for_block(&conn, block_audit_data.block_index())?;
+
+        Ok(BlockAuditDataResponse::new(block_audit_data, balances))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::db::test_utils::{append_and_sync, TestDbContext};
+    use mc_account_keys::AccountKey;
+    use mc_blockchain_types::{BlockContents, BlockVersion};
+    use mc_common::{
+        logger::{test_with_logger, Logger},
+        HashMap,
+    };
+    use mc_ledger_db::{
+        test_utils::{create_ledger, initialize_ledger},
+        Ledger,
+    };
+    use mc_transaction_core::TokenId;
+    use mc_transaction_core_test_utils::{
+        create_mint_config_tx_and_signers, create_mint_tx, create_test_tx_out,
+        mint_config_tx_to_validated as to_validated,
+    };
+
+    /// Create a test database with some data in it.
```

Comment:
It might be helpful to say here what the test data is intended to look like, or separate out the call where the data is added to make it easier to reference when looking at the tests.